src/cc65/expr.c

```diff
@@ -2501,6 +2501,37 @@ static void hie_internal (const GenDesc* Ops,   /* List of generators */
                 }
             }
 
+            /* Limit the calculated value to the range of its type */
```

Comment:
It looks like `+` and `-` are handled in `parseadd` and `parsesub`. Check everywhere `IVal` is assigned.

Reply:
Nice catch. Fix for arithmetic `+` and `-` upcoming.